Sources/Purchasing/Purchases.swift

```diff
@@ -1776,24 +1776,24 @@ extension Purchases: PurchasesOrchestratorDelegate {
     /**
      * Called when a user initiates a promotional in-app purchase from the App Store.
      *
-     * If your app is able to handle a purchase at the current time, run the deferment block in this method.
+     * If your app is able to handle a purchase at the current time, run the `startPurchase` block.
      *
-     * If the app is not in a state to make a purchase: cache the defermentBlock, then call the defermentBlock
+     * If the app is not in a state to make a purchase: cache the `startPurchase` block, then call it
      * when the app is ready to make the promotional purchase.
      *
-     * If the purchase should never be made, you don't need to ever call the defermentBlock and ``Purchases``
-     * will not proceed with promotional purchases.
+     * If the purchase should never be made, you don't need to ever call the `startPurchase` block
+     * and ``Purchases`` will not proceed with promotional purchases.
```

Comment:
i think we should update all `promotionals` -> `promoted`

Reply:
i know we have an example coming in MagicWeather...but considering how insanely confusing this flow is, I wonder if we should have an example here too? or perhaps point people to MagicWeather?

Reply:
Oh yeah good idea. Do you mean the example link to test this?

Reply:
i just meant an example of how a customer would implement this. but yeah a link to apple's docs for testing would also be helpful! with the URL and all

Reply:
You mean with actual code? We'd have to chose between `Swift` and `Objective-C` :/

Reply:
yeah true. i thought we had other examples elsewhere as precedence but maybe that was pre-migration. anyways, i think the documentation update should help enough! maybe we can point to the MagicWeather call from our site docs on the flow.

Reply:
> yeah true. i thought we had other examples elsewhere as precedence but maybe that was pre-migration. 

Oh I think we have some of that for the `AsyncStream` which is Swift only anyway.

>  i think the documentation update should help enough! maybe we can point to the MagicWeather call from our site docs on the flow.

I already added some of your doc suggestions 👍🏻 
I'll add the example link for how to test it and [this link](https://developer.apple.com/documentation/storekit/original_api_for_in-app_purchase/testing_promoted_in-app_purchases) (cc @aboedo any chance you could make a short link for that? 🙏🏻)